rpc_new/client.ts

```diff
@@ -26,13 +46,28 @@ export class Client<DiscoveryValue, SendE, InternalE, CloseE> {
         const pendingCall = this.pendingCalls[id]!;
         pendingCall.resolve(e);
         delete this.pendingCalls[id];
+        this.pendingSubscriptions[id]?.(e);
+        delete this.pendingSubscriptions[id];
+      }
+      for (const id in this.activeSubscriptions) {
+        this.activeSubscriptions[id]?.(e);
+        delete this.activeSubscriptions[id];
       }
     } else if (e.id) {
       const pendingCall = this.pendingCalls[e.id]!;
       pendingCall.resolve(e);
       delete this.pendingCalls[e.id];
+      if (this.pendingSubscriptions[e.id]) {
+        if (e.error) {
+          this.pendingSubscriptions[e.id]!(e);
+        } else {
+          this.activeSubscriptions[e.result] = this.pendingSubscriptions[e.id]!;
+          this.activeSubscriptionByMessageId[e.id] = e.result;
+        }
+        delete this.pendingSubscriptions[e.id];
+      }
```

Comment:
there are many `if`s here but for subscriptions we need to filter the initialization message so we can setup the listener for the actual subscription messages